client/implementation/src/main/java/io/smallrye/graphql/client/dynamic/ResponseImpl.java

```diff
@@ -63,4 +66,7 @@ public String toString() {
         return "GraphQLResponse{" + "data=" + data + ", errors=" + errors + '}';
     }
 
+    public List<Map.Entry<String, String>> getHeaders(){
+        return new ArrayList<>(headers);
```

Comment:
Maybe I'd prefer wrapping the headers into an unmodifiable collection in the constructor (they're not supposed to change after the constructor finishes), and here returning that already unmodifiable collection, instead of allocating a new object.. WDYT?

Reply:
Changed the constructor code as suggested.